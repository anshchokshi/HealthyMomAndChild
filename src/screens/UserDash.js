import React, { useEffect, useState, useContext} from 'react';
import { Button, StyleSheet, Text, TouchableOpacity, View, Modal, Pressable } from 'react-native'
import { firebase } from '../firebase'
import { Header } from '@rneui/themed'
import {LinearGradient} from 'expo-linear-gradient';
import { Image } from '@rneui/themed';
import { UserContext } from '../context/UserContext'
import { getWeeksDiff } from '../helpers/Date'

const UserDash = ({ navigation }) => {
    const auth = firebase.auth();
    const { userProfile } = useContext(UserContext)
    const handleBabyDev = () => {
            const lmpString = userProfile?.pregnantProfile?.LastMenstrualPeriod
            if (lmpString != null) {
                const lmp = new Date(lmpString)
                const weekNumber = getWeeksDiff(lmp, new Date())
                navigation.navigate("Fetal Screen", {weekNumber: Math.min(weekNumber, 42)})
            }
    }
    const handleSignOut = () => {
        auth.signOut().then(() => {
            navigation.replace("Login")
          })
          .catch(error => alert(error.message))
      }

    const handleWeightBP = () => {navigation.navigate("Summary")}

    const handleAppointments = () => {navigation.navigate("Appointment")}
    const handlerelatedWords = () => {navigation.navigate("related words screen")}
    const [modalVisible, setModalVisible] = useState(false);

    return(
        <View style={styles.container}>
            <View style={styles.headerContainer}>
                <LinearGradient
                    colors= {['#fa7878','#F08686', 'white']}
                    style= {styles.gradientHeaderStyle}
                    locations= {[0.1, 0.2, 1.0]}
                >
                    <Text style={styles.textMargin}>Your Dashboard</Text>
                </LinearGradient>
            </View>
            <View 
                style = {styles.dashContainer}>
                <Modal
                    animationType="slide"
                    transparent={true}
                    visible={modalVisible}
                        >

                <View style={styles.modalView}>
                    <Text style={styles.modalText}>This feature will be developed later.</Text>
                    <Pressable
                        style={styles.buttonClose}
                        onPress={() => setModalVisible(!modalVisible)}
                        >
                        <Text style={{fontSize:17}}>Hide</Text>
                        </Pressable>
                        <Pressable
                        style={styles.buttonClose}
                        onPress={handleSignOut}
                        >
                        <Text style={{fontSize:17}}>SignOut</Text>
                        </Pressable>
                    </View>
        
                </Modal>
                {/* <Text>MainPage</Text> */}
                <View style={styles.buttonCont}>
                    <TouchableOpacity style={styles.buttonLeft}
                        onPress={handleBabyDev}
                        testID="baby-development">
                        <Image source={require('../assets/BD.png')} style={styles.buttonImg} blurRadius={5}></Image>
                        <Text style={styles.buttonText}>Baby Development</Text>
                    </TouchableOpacity>
                    <TouchableOpacity style={styles.buttonRight}
                        onPress={handleWeightBP}
                        testID="weight-gain">
                    <Image source={require('../assets/WBP.png')} style={styles.buttonImg} blurRadius={5}></Image>
                        <Text style={styles.buttonText}>Your Weight Gain and Blood Pressure</Text>
                    </TouchableOpacity>
                </View>
                <View style={styles.buttonCont}>
                    <TouchableOpacity style={styles.buttonLeft}
                        onPress={handleAppointments}
                        testID="appointments">
                        <Image source={require('../assets/APTS.png')} style={styles.buttonImg} blurRadius={5}></Image>
                        <Text style={styles.buttonText}>Your Appointments</Text>
                    </TouchableOpacity>
                    <TouchableOpacity style={styles.buttonRight}
                    onPress={() => setModalVisible(true)}
                    >
                    <Image source={require('../assets/SA.png')} style={styles.buttonImg} blurRadius={5}></Image>
                        <Text style={styles.buttonText}>Surveys and Articles</Text>
                    </TouchableOpacity>
                </View>
                <View style={styles.buttonContCenter}>
                    <TouchableOpacity style={styles.buttonCenter}
                    onPress={handlerelatedWords}
                    testID="related-words">
                    <Image source={require('../assets/EMC.png')} style={styles.buttonImg} blurRadius={5}></Image>
                        <Text style={styles.buttonText}>Educational Medical Content</Text>
                    </TouchableOpacity>
                </View>
            </View>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        // alignItems: 'center',
        backgroundColor:'#ffffff',
      },
      gradientHeaderStyle :{
        width: "100%",
        height: "100%",
        borderRadius: "25px"
      },
      headerContainer :{
        width: "100%",
        height: "100%",
        backgroundColor: "#F08686",
      },
      dashContainer :{
          position: "absolute",
          marginTop: "30%",
        //   marginLeft: "3%",
        //   marginRight: "3%",
          width: "100%",
          height: "80%",
          backgroundColor: 'rgba(250, 250, 250, 0.35)',
          borderRadius: "25px",
      },
      textMargin :{
          marginTop: "15%",
          marginLeft: "3%",
          fontSize: "40px",
          color: "#ffffff",
          fontWeight:"bold",
      },
      buttonCont: {
          width: "100%",
          height: "33%",
          marginBottom: "2%",
        //   position: "absolute",
          flexDirection: "row",
        //   backgroundColor: "#123343"
      },
      buttonLeft: {
          width: "40%",
          height: "100%",
          marginTop: "2%",
          marginBottom: "2%",
          marginLeft: "3%",
          marginRight: "3%",
          position: "relative",
          left: 0,
        //   backgroundColor: "rgba(250, 250, 250, 0.5)",
          borderRadius: "25%",
          flex: 1
      },
      buttonRight: {
        width: "40%",
        height: "100%",
        marginTop: "2%",
        marginBottom: "2%",
        // marginLeft: "3%",
        marginRight: "3%",
        // position: "relative",
        // backgroundColor: "rgba(250, 250, 250, 0.5)",
        // right: 0,
        borderRadius: "25%",
        flex: 1
    },
    buttonContCenter: {
        alignItems: "center",
        width: "100%",
        height: "33%",
        marginBottom: "2%"
      //   position: "absolute",
        // flexDirection: "row",
        // backgroundColor: "#123343"
    },
    buttonCenter: {
        width: "40%",
        height: "100%",
        marginTop: "2%",
        marginBottom: "2%",
        // marginLeft: "3%",
        // marginRight: "3%",
        // position: "relative",
        // backgroundColor: "#000000",
        // right: 0,
        borderRadius: "25%",
        flex: 1
    },
    buttonText: {
        color:"#000000",
        fontSize: "20px",
        fontWeight: 'bold',
        // textAlign: "left",
        marginTop: "25%",
        marginLeft: "5%",
        position: 'absolute',
        zIndex: 1
    },
    buttonTextWhite: {
        color:"#ffffff",
        fontSize: "20px",
        fontWeight: 'bold',
        // textAlign: "left",
        marginTop: "25%",
        marginLeft: "5%",
        position: 'absolute',
        zIndex: 1
    },
    buttonImg: {
        // resizeMode: 'contain',
        // flex: 2
        width: '100%',
        height: '100%',
        borderRadius: '25%',
        // flex: 1
        // position:'absolute'
    },
    abs: {
        // position: 'absolute'
    },
    modalView: {
        marginTop: '80%',
        backgroundColor: "white",
        borderRadius: 20,
        padding: 35,
        alignItems: "center",
        shadowColor: "#000",
        shadowOffset: {
          width: 0,
          height: 2
        },
        shadowOpacity: 0.25,
        shadowRadius: 4,
        elevation: 5
      },
      modalText: {
        marginBottom: 15,
        textAlign: "center",
        fontSize:18,
        fontWeight:'700'
      },
      buttonClose: {
        backgroundColor: "#2196F3",
        borderRadius: 10,
        padding: 15,
        elevation: 2
      },
    
})

export default UserDash