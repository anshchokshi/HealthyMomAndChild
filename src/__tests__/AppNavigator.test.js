import React from 'react';
import {create} from 'react-test-renderer'
import { fireEvent, render } from "@testing-library/react-native"
import UserDash from '../screens/UserDash';


describe('User Dash', () => {
  const dash = create(<UserDash />)
  

  it('User Dash is rendered correctly', () => {
    expect(dash.toJSON()).toMatchSnapshot()
    
  });
  it('Navigate to fetal screen', () => {
    const navigate = jest.fn();
    const { getByTestId } = render(<UserDash navigation={{navigate}}/>);
    fireEvent.press(getByTestId('baby-development'));
    expect(navigate).toHaveBeenCalledWith('Fetal Screen');
  });

  it('Navigate to weight gain and blood pressure screen', () => {
    const navigate = jest.fn();
    const { getByTestId } = render(<UserDash navigation={{navigate}}/>);
    fireEvent.press(getByTestId('weight-gain'));
    expect(navigate).toHaveBeenCalledWith('Summary');
  });

  it('Navigate to appointments screen', () => {
    const navigate = jest.fn();
    const { getByTestId } = render(<UserDash navigation={{navigate}}/>);
    fireEvent.press(getByTestId('appointments'));
    expect(navigate).toHaveBeenCalledWith('Appointment');
  });

  it('Navigate to related words screen', () => {
    const navigate = jest.fn();
    const { getByTestId } = render(<UserDash navigation={{navigate}}/>);
    fireEvent.press(getByTestId('related-words'));
    expect(navigate).toHaveBeenCalledWith('related words screen');
  });
});
