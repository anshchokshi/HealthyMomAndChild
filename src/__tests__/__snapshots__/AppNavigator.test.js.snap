// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Dash User Dash is rendered correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#F08686",
        "height": "30%",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": "40px",
          "fontWeight": "bold",
          "marginLeft": "3%",
          "marginTop": "5%",
        }
      }
    >
      Your Dashboard
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderRadius": "25px",
        "height": "80%",
        "marginLeft": "3%",
        "marginRight": "3%",
        "marginTop": "20%",
        "position": "absolute",
        "width": "95%",
      }
    }
  >
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      transparent={true}
      visible={false}
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "height": "33%",
          "marginBottom": "2%",
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderRadius": "25%",
            "flex": 1,
            "height": "100%",
            "left": 0,
            "marginBottom": "2%",
            "marginLeft": "3%",
            "marginRight": "3%",
            "marginTop": "2%",
            "opacity": 1,
            "position": "relative",
            "width": "40%",
          }
        }
        testID="baby-development"
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            Object {
              "backgroundColor": "transparent",
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <Image
            blurRadius={5}
            onLoad={[Function]}
            source={1}
            style={
              Object {
                "borderRadius": "25%",
                "bottom": 0,
                "height": "100%",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": "100%",
              }
            }
            testID="RNE__Image"
            theme={
              Object {
                "colors": Object {
                  "background": "#ffffff",
                  "black": "#242424",
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "grey": "rgba(0, 0, 0, 0.54)",
                      "primary": "#2196f3",
                      "searchBg": "#dcdce1",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "default": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "web": Object {
                      "error": "#ff190c",
                      "grey": "#393e42",
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ad1457",
                  "success": "#52c41a",
                  "warning": "#faad14",
                  "white": "#ffffff",
                },
                "spacing": Object {
                  "lg": 12,
                  "md": 8,
                  "sm": 4,
                  "xl": 24,
                  "xs": 2,
                },
              }
            }
            transitionDuration={360}
          />
          <View
            accessibilityElementsHidden={true}
            collapsable={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#bdbdbd",
                  "borderRadius": "25%",
                  "height": "100%",
                  "justifyContent": "center",
                  "width": "100%",
                }
              }
              testID="RNE__Image__placeholder"
            />
          </View>
          <View
            style={
              Object {
                "borderRadius": "25%",
                "height": "100%",
                "width": "100%",
              }
            }
            testID="RNE__Image__children__container"
          />
        </View>
        <Text
          style={
            Object {
              "color": "#000000",
              "fontSize": "20px",
              "fontWeight": "bold",
              "marginLeft": "5%",
              "marginTop": "25%",
              "position": "absolute",
              "zIndex": 1,
            }
          }
        >
          Baby Development
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderRadius": "25%",
            "flex": 1,
            "height": "100%",
            "marginBottom": "2%",
            "marginRight": "3%",
            "marginTop": "2%",
            "opacity": 1,
            "width": "40%",
          }
        }
        testID="weight-gain"
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            Object {
              "backgroundColor": "transparent",
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <Image
            blurRadius={5}
            onLoad={[Function]}
            source={1}
            style={
              Object {
                "borderRadius": "25%",
                "bottom": 0,
                "height": "100%",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": "100%",
              }
            }
            testID="RNE__Image"
            theme={
              Object {
                "colors": Object {
                  "background": "#ffffff",
                  "black": "#242424",
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "grey": "rgba(0, 0, 0, 0.54)",
                      "primary": "#2196f3",
                      "searchBg": "#dcdce1",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "default": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "web": Object {
                      "error": "#ff190c",
                      "grey": "#393e42",
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ad1457",
                  "success": "#52c41a",
                  "warning": "#faad14",
                  "white": "#ffffff",
                },
                "spacing": Object {
                  "lg": 12,
                  "md": 8,
                  "sm": 4,
                  "xl": 24,
                  "xs": 2,
                },
              }
            }
            transitionDuration={360}
          />
          <View
            accessibilityElementsHidden={true}
            collapsable={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#bdbdbd",
                  "borderRadius": "25%",
                  "height": "100%",
                  "justifyContent": "center",
                  "width": "100%",
                }
              }
              testID="RNE__Image__placeholder"
            />
          </View>
          <View
            style={
              Object {
                "borderRadius": "25%",
                "height": "100%",
                "width": "100%",
              }
            }
            testID="RNE__Image__children__container"
          />
        </View>
        <Text
          style={
            Object {
              "color": "#000000",
              "fontSize": "20px",
              "fontWeight": "bold",
              "marginLeft": "5%",
              "marginTop": "25%",
              "position": "absolute",
              "zIndex": 1,
            }
          }
        >
          Your Weight Gain and Blood Pressure
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "height": "33%",
          "marginBottom": "2%",
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderRadius": "25%",
            "flex": 1,
            "height": "100%",
            "left": 0,
            "marginBottom": "2%",
            "marginLeft": "3%",
            "marginRight": "3%",
            "marginTop": "2%",
            "opacity": 1,
            "position": "relative",
            "width": "40%",
          }
        }
        testID="appointments"
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            Object {
              "backgroundColor": "transparent",
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <Image
            blurRadius={5}
            onLoad={[Function]}
            source={1}
            style={
              Object {
                "borderRadius": "25%",
                "bottom": 0,
                "height": "100%",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": "100%",
              }
            }
            testID="RNE__Image"
            theme={
              Object {
                "colors": Object {
                  "background": "#ffffff",
                  "black": "#242424",
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "grey": "rgba(0, 0, 0, 0.54)",
                      "primary": "#2196f3",
                      "searchBg": "#dcdce1",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "default": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "web": Object {
                      "error": "#ff190c",
                      "grey": "#393e42",
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ad1457",
                  "success": "#52c41a",
                  "warning": "#faad14",
                  "white": "#ffffff",
                },
                "spacing": Object {
                  "lg": 12,
                  "md": 8,
                  "sm": 4,
                  "xl": 24,
                  "xs": 2,
                },
              }
            }
            transitionDuration={360}
          />
          <View
            accessibilityElementsHidden={true}
            collapsable={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#bdbdbd",
                  "borderRadius": "25%",
                  "height": "100%",
                  "justifyContent": "center",
                  "width": "100%",
                }
              }
              testID="RNE__Image__placeholder"
            />
          </View>
          <View
            style={
              Object {
                "borderRadius": "25%",
                "height": "100%",
                "width": "100%",
              }
            }
            testID="RNE__Image__children__container"
          />
        </View>
        <Text
          style={
            Object {
              "color": "#000000",
              "fontSize": "20px",
              "fontWeight": "bold",
              "marginLeft": "5%",
              "marginTop": "25%",
              "position": "absolute",
              "zIndex": 1,
            }
          }
        >
          Your Appointments
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderRadius": "25%",
            "flex": 1,
            "height": "100%",
            "marginBottom": "2%",
            "marginRight": "3%",
            "marginTop": "2%",
            "opacity": 1,
            "width": "40%",
          }
        }
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            Object {
              "backgroundColor": "transparent",
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <Image
            blurRadius={5}
            onLoad={[Function]}
            source={1}
            style={
              Object {
                "borderRadius": "25%",
                "bottom": 0,
                "height": "100%",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": "100%",
              }
            }
            testID="RNE__Image"
            theme={
              Object {
                "colors": Object {
                  "background": "#ffffff",
                  "black": "#242424",
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "grey": "rgba(0, 0, 0, 0.54)",
                      "primary": "#2196f3",
                      "searchBg": "#dcdce1",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "default": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "web": Object {
                      "error": "#ff190c",
                      "grey": "#393e42",
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ad1457",
                  "success": "#52c41a",
                  "warning": "#faad14",
                  "white": "#ffffff",
                },
                "spacing": Object {
                  "lg": 12,
                  "md": 8,
                  "sm": 4,
                  "xl": 24,
                  "xs": 2,
                },
              }
            }
            transitionDuration={360}
          />
          <View
            accessibilityElementsHidden={true}
            collapsable={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#bdbdbd",
                  "borderRadius": "25%",
                  "height": "100%",
                  "justifyContent": "center",
                  "width": "100%",
                }
              }
              testID="RNE__Image__placeholder"
            />
          </View>
          <View
            style={
              Object {
                "borderRadius": "25%",
                "height": "100%",
                "width": "100%",
              }
            }
            testID="RNE__Image__children__container"
          />
        </View>
        <Text
          style={
            Object {
              "color": "#000000",
              "fontSize": "20px",
              "fontWeight": "bold",
              "marginLeft": "5%",
              "marginTop": "25%",
              "position": "absolute",
              "zIndex": 1,
            }
          }
        >
          Surveys and Articles
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": "33%",
          "marginBottom": "2%",
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderRadius": "25%",
            "flex": 1,
            "height": "100%",
            "marginBottom": "2%",
            "marginTop": "2%",
            "opacity": 1,
            "width": "40%",
          }
        }
        testID="related-words"
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            Object {
              "backgroundColor": "transparent",
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <Image
            blurRadius={5}
            onLoad={[Function]}
            source={1}
            style={
              Object {
                "borderRadius": "25%",
                "bottom": 0,
                "height": "100%",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": "100%",
              }
            }
            testID="RNE__Image"
            theme={
              Object {
                "colors": Object {
                  "background": "#ffffff",
                  "black": "#242424",
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "grey": "rgba(0, 0, 0, 0.54)",
                      "primary": "#2196f3",
                      "searchBg": "#dcdce1",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "default": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "web": Object {
                      "error": "#ff190c",
                      "grey": "#393e42",
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ad1457",
                  "success": "#52c41a",
                  "warning": "#faad14",
                  "white": "#ffffff",
                },
                "spacing": Object {
                  "lg": 12,
                  "md": 8,
                  "sm": 4,
                  "xl": 24,
                  "xs": 2,
                },
              }
            }
            transitionDuration={360}
          />
          <View
            accessibilityElementsHidden={true}
            collapsable={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#bdbdbd",
                  "borderRadius": "25%",
                  "height": "100%",
                  "justifyContent": "center",
                  "width": "100%",
                }
              }
              testID="RNE__Image__placeholder"
            />
          </View>
          <View
            style={
              Object {
                "borderRadius": "25%",
                "height": "100%",
                "width": "100%",
              }
            }
            testID="RNE__Image__children__container"
          />
        </View>
        <Text
          style={
            Object {
              "color": "#000000",
              "fontSize": "20px",
              "fontWeight": "bold",
              "marginLeft": "5%",
              "marginTop": "25%",
              "position": "absolute",
              "zIndex": 1,
            }
          }
        >
          Educational Medical Content
        </Text>
      </View>
    </View>
  </View>
</View>
`;
